Hymns/Infrastructure/Analytics/AnalyticsLogger.swift

```diff
@@ -1,33 +1,48 @@
 import FirebaseAnalytics
 import Foundation
+import Resolver
 
 /**
  * Wrapper for `FirebaseAnalytics` to help keep track of what we are logging and analyzing.
  */
 class AnalyticsLogger {
 
+    private let backgroundThread: DispatchQueue
+
+    init(backgroundThread: DispatchQueue = Resolver.resolve(name: "background")) {
+        self.backgroundThread = backgroundThread
+    }
```

Comment:
Why do you use custom dispatchQueues for background instead of the DispatchQueue.global? I am guessing it is better for dependency injection?

Reply:
I don't use a custom dispatch queue. If you look at https://github.com/HymnalDreamTeam/Hymns-iOS/blob/master/Hymns/Infrastructure/Dependency%20Injection/AppDelegate%2BInjection.swift#L10, I'm merely injecting the DispatchQueue.global into here, so it is using the global dispatch queue